cirq/ops/common_gates.py

```diff
@@ -103,6 +103,18 @@ def _act_on_(self, args: Any):
                 tableau.xs[:, q] ^= tableau.zs[:, q]
             return True
 
+        if isinstance(args, clifford.ActOnStabilizerCHFormArgs):
+            if protocols.is_parameterized(self) or self.exponent % 0.5 != 0:
+                return NotImplemented
+            if H._act_on_(args) is True:
```

Comment:
I would replace these with explicit calls to the methods of ChForm, or just assert that they succeeded.

Reply:
The methods of the ChForm are to be deleted as per #2948.